haystack/nodes/_json_schema.py

```diff
@@ -380,6 +380,9 @@ def get_json_schema(filename: str, version: str, modules: List[str] = ["haystack
         ],
         "definitions": schema_definitions,
     }
+
+    # Leveraging an implementation detail of dict: keys stay in the order they are inserted.
```

Comment:
Some might say that using an OrderedDict instead of a Dict makes this more explicit. However, I am okay with keeping it just as it is right now. 👍

Reply:
You have a point :smile: Somehow I totally forgot about `OrderedDict`. Fixing!

Reply:
Somehow `OrderedDict` doesn't seem to work! I'm really surprised but I won't investigate right now. Reverting to `dict` Really odd...